### PR TITLE
Amit_Enemy_Animations_And_Stats_Scripts_And_AnimationClips

### PR DESCRIPTION
**Pull request by TheWazzzzzup**

### CODE REVIEW

File: Defuq/Assets/Scripts/EnemyBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class EnemyBehavior : MonoBehaviour
```
Comment (GDIVX) [inline Defuq/Assets/Scripts/EnemyBehavior.cs:6]: This class have too many respoinsibilities and will hard to maintian over time. I would recommend spliting the code to 3 diffrent classes: one for animation, one scriptableObject for data and one for behavoiur.

File: Defuq/Assets/Scripts/ScriptableObjectScripts/EnemyStatsSO.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+
+[CreateAssetMenu(fileName = "Enemy",menuName = "Create New Enemy SO",order = 1)]
+public class EnemyStatsSO : ScriptableObject
```
Comment (GDIVX) [inline Defuq/Assets/Scripts/ScriptableObjectScripts/EnemyStatsSO.cs:8]: You already got this script in another folder, did you forgot to delete this one?

**Review by GDIVX — CHANGES_REQUESTED** [review]

File: Defuq/Assets/Scripts/EnemyBehavior.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        enemyAnimator.SetBool(AttackBoolExpersion, true);
+    }
+
+    public void AnimatorAttackFalse()
+    {
+        enemyAnimator.SetBool(AttackBoolExpersion, false);
+    }
+
+    private void Start()
```
Comment (GDIVX) [inline Defuq/Assets/Scripts/EnemyBehavior.cs]: Try using Awake(), it is called at compellation while Start() run in runtime.

File: Defuq/Assets/Scripts/EnemyBehavior.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        else if (currentHealth <= 0)
+        {
+            currentHealth = 0;
+            enemyKilled.Invoke();
+            Debug.Log("Enemy Killed");
+        }
+    }
+
+
+    // Used to debug Animations, Attack will Initiate, and killed will initate evantualy
```
Comment (GDIVX) [inline Defuq/Assets/Scripts/EnemyBehavior.cs]: remove debug code before pull request

**Review by GDIVX — CHANGES_REQUESTED** [review]

File: Defuq/Assets/Scripts/ScriptableObjectScripts/EnemyStatsSO.cs (hunk 2 above)
Comment (TheWazzzzzup) [reply Defuq/Assets/Scripts/ScriptableObjectScripts/EnemyStatsSO.cs:8]: where?

File: Defuq/Assets/Scripts/EnemyBehavior.cs (hunk 1 above)
Comment (TheWazzzzzup) [reply Defuq/Assets/Scripts/EnemyBehavior.cs:6]: would you recommend splitting it into 3 scripts or using different classes inside the same script?

Comment (TheWazzzzzup) [reply Defuq/Assets/Scripts/EnemyBehavior.cs:6]: If you took a look now you would see I have split it into 3 different scripts

File: Defuq/Assets/Scripts/ScriptableObjectScripts/EnemyStatsSO.cs (hunk 2 above)
Comment (TheWazzzzzup) [reply Defuq/Assets/Scripts/ScriptableObjectScripts/EnemyStatsSO.cs:8]: I only have one scriptable object script...

**Review by GDIVX — CHANGES_REQUESTED** [review]
Your branch seems to be coruopted and dose not ignore library files. I would highly reccomned copying your scripts to a new branch and to delete this one. Whatever you do, DO NOT merge this branch to main.

**Pull request closed without merging**